### PR TITLE
enh(cenreonUi/MenuSkeleton):  header and menu skeleton creates an unstable UI

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
update css menu skeleton

### CODE REVIEW

File: packages/centreon-ui/src/MenuSkeleton/index.tsx [hunk 1]
```diff
@@ -1,31 +1,50 @@
-import { useTheme, alpha } from '@mui/material';
+import clsx from 'clsx';
+import { equals } from 'ramda';
+
+import { SkeletonProps, useTheme, alpha } from '@mui/material';
 import makeStyles from '@mui/styles/makeStyles';
 
-import LoadingSkeleton from '../LoadingSkeleton';
+import { LoadingSkeleton } from '@centreon/ui';
```
Comment (Thebarda) [inline packages/centreon-ui/src/MenuSkeleton/index.tsx:7]:
```suggestion
import LoadingSkeleton from '../LoadingSkeleton';
```

File: packages/centreon-ui/src/MenuSkeleton/index.tsx [hunk 2]
```diff
@@ -1,31 +1,50 @@
...
+      className={clsx(classes.skeleton, className)}
+      height={
+        equals(typeof height, 'string')
+          ? height
+          : theme.spacing(height as number)
+      }
+      variant={variant}
+      width={
+        equals(typeof width, 'string') ? width : theme.spacing(width as number)
+      }
```
Comment (Thebarda) [inline packages/centreon-ui/src/MenuSkeleton/index.tsx:47]: We want to define width and height using the default Material UI spacing
```suggestion
      height={
        theme.spacing(height as number)
      }
      variant={variant}
      width={
        theme.spacing(width as number)
      }
```

File: packages/centreon-ui/src/MenuSkeleton/index.tsx [hunk 3]
```diff
@@ -1,31 +1,50 @@
...
   },
 }));
 
 interface Props {
   animate?: boolean;
-  width?: number;
+  className?: string;
+  height?: number | string;
+  variant?: SkeletonProps['variant'];
+  width?: number | string;
```
Comment (Thebarda) [inline packages/centreon-ui/src/MenuSkeleton/index.tsx:22]:
```suggestion
  height?: number;
  variant?: SkeletonProps['variant'];
  width?: number;
```

**Review by Thebarda — APPROVED** [review]

File: packages/centreon-ui/src/MenuSkeleton/index.tsx [hunk 4]
```diff
@@ -1,31 +1,50 @@
...
   return (
     <LoadingSkeleton
       animation={animate ? 'wave' : false}
-      className={classes.skeleton}
-      height={theme.spacing(5)}
-      variant="text"
-      width={theme.spacing(width)}
+      className={clsx(classes.skeleton, className)}
+      height={
+        equals(typeof height, 'string')
```
Comment (bdauria) [inline packages/centreon-ui/src/MenuSkeleton/index.tsx:40]: Wouldn't it make more sense to accept just one type for height?

I don't think we need that flexibility

File: packages/centreon-ui/src/MenuSkeleton/index.tsx [hunk 5]
```diff
@@ -1,31 +1,50 @@
...
+          ? height
+          : theme.spacing(height as number)
+      }
+      variant={variant}
+      width={
+        equals(typeof width, 'string') ? width : theme.spacing(width as number)
```
Comment (bdauria) [inline packages/centreon-ui/src/MenuSkeleton/index.tsx:46]: Same remark here

**Review by Thebarda — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Pull request merged**